website/src/config-reference.md

```diff
@@ -457,6 +457,38 @@ parameters:
 
 When set to `true`, PHPStan is strict about values with an unspecified (implicit `mixed`) type. It enables the same checks for values with no type specified that rule level 9 enables for explicitly specified `mixed` type values.
 
+### `checkBenevolentUnionTypes`
+
+<div class="text-xs inline-block border border-green-600 text-green-600 bg-green-100 rounded px-1 mb-4">Available in PHPStan 1.9.0</div>
+
+**default**: `false`
+
+PHPStan has defined some benevolent union types. Those unions are not checked by the `checkUnionTypes` option enabled on level 7:
```

Comment:
does the documentation mention anywhere what a "benevolent union type" actually is? if so it would be great to link to it here.

Reply:
```suggestion
PHPStan defines some benevolent union types, such as `array-key`. Benevolent unions are ignored by the `checkUnionTypes` option enabled on level 7:
```

Reply:
> does the documentation mention anywhere what a "benevolent union type" actually is? if so it would be great to link to it here.

There is none

Reply:
this is one of the big open mysteries :D the best we have so far is this https://github.com/phpstan/phpstan/discussions/6440